python/test/test_find_pypi_tag.py

```diff
@@ -0,0 +1,66 @@
+# To run: `python -m unittest test_find_pypi_tag.py`
+
+import unittest
+import sys
+from packaging import version
+sys.path.append('..')
+from find_pypi_tag import compute_appropriate_version
+
+class TestFindTestPyPiTag(unittest.TestCase):
+
+    def compare(self, releases: list[str], v: str, expected_v: str):
+        """
+        Test helper, to avoid repeating myself
+        """
+        new_v = compute_appropriate_version(
+            current_v=version.parse(v),
+            releases=[version.parse(v) for v in releases],
+            current=False
+        )
+        self.assertEqual(version.parse(expected_v), new_v)
+
+    def test_when_not_exist_with_prereleasetag(self):
+        self.compare(releases=["3.2.0"], v="3.2.1-alpha", expected_v="3.2.1a0")
```

Comment:
I had to add an actual unit test to do TDD for the find_pypi_tag.py file, I wasn't managing to do it without it. This shows pretty well how it functions.